packages/repo-tools/src/commands/openapi/client/generate.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import chalk from 'chalk';
+import { resolve } from 'path';
+import { OPENAPI_IGNORE_FILES, OUTPUT_PATH } from '../constants';
+import { promisify } from 'util';
+import { exec as execCb } from 'child_process';
+import { paths as cliPaths } from '../../../lib/paths';
+import { mkdirpSync } from 'fs-extra';
+import fs from 'fs-extra';
+
+const exec = promisify(execCb);
+
+async function generate(spec: string, outputDirectory: string) {
+  const resolvedOpenapiPath = resolve(spec);
+  const resolvedOutputDirectory = resolve(outputDirectory, OUTPUT_PATH);
+  mkdirpSync(resolvedOutputDirectory);
+
+  await fs.mkdirp(resolvedOutputDirectory);
+
+  await fs.writeFile(
+    resolve(resolvedOutputDirectory, '.openapi-generator-ignore'),
+    OPENAPI_IGNORE_FILES.join('\n'),
+  );
+
+  await exec(
+    `cd ${cliPaths.ownDir} && yarn openapi-generator-cli generate -i "${resolvedOpenapiPath}" -o "${resolvedOutputDirectory}" -g typescript -c "templates/typescript-backstage.yaml"`,
```

Comment:
Use the `cwd` option rather than `cd`. I'd say it's also nicer to look up the `openapi-generator-cli` bin path rather than calling it via `yarn`. Prefer using `execFile` or similar as well, one where you can pass an array of parameters rather than a full command string. Helps avoid escaping issues.